openmc/material.py

```diff
@@ -413,6 +413,8 @@ def add_nuclide(self, nuclide, percent, percent_type='ao'):
             Z, _, _ = openmc.data.zam(nuclide)
         except ValueError as e:
             warnings.warn(str(e))
+        except KeyError as e:
```

Comment:
So it looks like KeyError is raised because the regular expression matching works but then it fails on `ATOMIC_NUMBER[symbol]`. I think what we ought to do is add a check that `symbol in ATOMIC_NUMBER` inside of `zam` and raise a different `ValueError` saying that the atomic symbol is not recognized. That seems more consistent to me; a ValueError is raised if the argument to `zam` is not a valid value (regardless of how it is invalid). And, it means you won't need to change any core here in material.py.